src/Tizen.NUI/src/public/DragAndDrop/DragAndDrop.cs

```diff
@@ -76,79 +101,74 @@ public void StartDragAndDrop(View sourceView, View shadowView, DragData dragData
-                {
-                    BackgroundColor = Color.Transparent,
-                };
-            }
 
-            //Initialize Drag Window Position and Size based on Shadow View Position and Size
-            mDragWindow.SetPosition(new Position2D((int)shadowView.Position.X, (int)shadowView.Position.Y));
-            mDragWindow.SetWindowSize(new Size(shadowWidth, shadowHeight));
+                shadowWidth = (int)shadowView.Size.Width;
+                shadowHeight = (int)shadowView.Size.Height;
 
-            //Make Shadow View Transparent
-            shadowView.SetOpacity(0.9f);
-
-            //Make Position 0, 0 for Moving into Drag Window
-            shadowView.Position = new Position(0, 0);
+                // Prevents shadowView size from being smaller than 100 pixel
+                if (shadowView.Size.Width < 100)
+                {
+                    shadowWidth = 100;
+                }
 
-            if (mShadowView)
-            {
-                mShadowView.Hide();
-                mDragWindow.Remove(mShadowView);
-                mShadowView.Dispose();
-            }
+                if (shadowView.Size.Height < 100)
+                {
+                    shadowHeight = 100;
+                }
 
-            mShadowView = shadowView;
-            mDragWindow.Add(mShadowView);
+                //Initialize Drag Window Position and Size based on Shadow View Position and Size
+                mDragWindow.SetPosition(new Position2D((int)shadowView.Position.X, (int)shadowView.Position.Y));
+                mDragWindow.SetWindowSize(new Size(shadowWidth, shadowHeight));
+
+                //Make Shadow View Transparent
+                shadowView.SetOpacity(0.9f);
+
+                //Make Position 0, 0 for Moving into Drag Window
+                shadowView.Position = new Position(0, 0);
+            
+                mShadowView = shadowView;
+                mDragWindow.Add(mShadowView);
+           
+                sourceEventCb = (sourceEventType) =>
+                {   
+                    if ((DragSourceEventType)sourceEventType != DragSourceEventType.Start)
+                    {     
+                        Tizen.Log.Fatal("NUI", "DnD Source Event is Called");  
+                        ReleaseDragWindow();                
+                    }
 
-            //Update Window Directly
-            mDragWindow.VisibiltyChangedSignalEmit(true);
-            mDragWindow.RenderOnce();
+                    callback((DragSourceEventType)sourceEventType);
+                };
 
-            sourceEventCb = (sourceEventType) =>
-            {
-                if ((DragSourceEventType)sourceEventType == DragSourceEventType.Finish)
+                if (!Interop.DragAndDrop.StartDragAndDrop(SwigCPtr, View.getCPtr(sourceView), Window.getCPtr(mDragWindow), dragData.MimeType, dragData.Data,
```

Comment:
The "StartDragandDrop" API is a request for the Display server to allow "mDragWindow" to belong to the cursor.
After this API call, any change to "mDragWindow" can change the position of "mDragWindow".
For prevent mis-positioning of the mDragWindow, It's better that called after the mDragWindow.Show().

Reply:
modified the codes following this guide